hive/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableOperations.java

```diff
@@ -139,7 +140,7 @@ private DirectoryInfo getDirectoryInfo() {
     }
   }
 
-  private List<DirectoryInfo> getDirectoryInfosByFilter(Expression expression) {
+  private List<DirectoryInfo> getDirectoryInfosByFilter(Expression expression, PartitionSpec spec) {
```

Comment:
Any reason to pass the spec here instead of just using current().spec()?

Reply:
Fixed